applications/asset_tracker_v2/src/main.c

```diff
@@ -174,41 +174,48 @@ static void sub_state_set(enum sub_state_type new_state)
 }
 
 #if defined(CONFIG_NRF_MODEM_LIB)
-/* Check the return code from nRF modem library initialization to ensure that
- * the modem is rebooted if a modem firmware update is ready to be applied or
- * an error condition occurred during firmware update or library initialization.
- */
-static void modem_init(void)
-{
-	int ret = nrf_modem_lib_init();
 
-	/* Handle return values relating to modem firmware update */
-	switch (ret) {
-	case 0:
-		/* Initialization successful, no action required. */
-		return;
+static void on_modem_lib_dfu(int dfu_res, void *ctx)
+{
+	switch (dfu_res) {
 	case NRF_MODEM_DFU_RESULT_OK:
-		LOG_DBG("MODEM UPDATE OK. Will run new modem firmware after reboot");
+		LOG_DBG("nRF Modem firmware update succeeded");
```

Comment:
I would keep this line the same, just in case there are tests looking for it; the other prints are unchanged too.

Reply:
That would not be correct though, as the modem is already initialized now. But I will change the prefix.